modules/moderation.py

```diff
@@ -148,6 +184,57 @@ async def find_discord_tokens(self, message: discord.Message) -> None:
         )
         await message.reply(msg)
 
+    @commands.Cog.listener()
+    async def on_papi_dpy_modlog(self, payload: ModLogType, /) -> None:
+        moderation_payload = payload["payload"]
+        moderation_event = core.DiscordPyModerationEvent(moderation_payload["moderation_event_type"])
+
+        embed = discord.Embed(
+            title=f"Discord.py Moderation Event: {moderation_event.name.title()}",
+            colour=random_pastel_colour(),
+        )
+
+        target_id = moderation_payload["target_id"]
+        target = await self.bot.get_or_fetch_user(target_id)
+
+        moderation_reason = moderation_payload["reason"]
+
+        moderator_id = moderation_payload["author_id"]
+        moderator = self.dpy_mod_cache.get(moderator_id) or await self.bot.get_or_fetch_user(
+            moderator_id, cache=self.dpy_mod_cache
+        )
+
+        if moderator:
+            self.dpy_mod_cache[moderator.id] = moderator
+            moderator_format = f"{moderator.name} {PROSE_LOOKUP[moderation_event.value]} "
+            embed.set_author(name=moderator.name, icon_url=moderator.display_avatar.url)
+        else:
+            moderator_format = f"Unknown Moderator with ID: {moderator_id} {PROSE_LOOKUP[moderation_event.value]} "
+            embed.set_author(name=f"Unknown Moderator.")
+
+        if target:
+            target_format = target.name
+            embed.set_footer(text=f"{target.name} | {target_id}", icon_url=target.display_avatar.url)
+        else:
+            target_format = f"An unknown user with ID {target_id}"
+            embed.set_footer(text=f"Not Found | {target_id}")
+        embed.add_field(name="Reason", value=moderation_reason or "No reason given.")
+
+        embed.description = moderator_format + target_format
+
+        when = datetime.datetime.fromisoformat(moderation_payload["event_time"])
+        embed.timestamp = when
+
+        guild = self.bot.get_guild(490948346773635102)
+        assert guild
+
+        channel = guild.get_channel(Channels.DPY_MOD_LOGS)
+        assert isinstance(channel, discord.TextChannel)  # This is static
+
+        view = ModerationRespostView(timeout=180, target_id=target_id, target_reason=moderation_reason)
```

Comment:
```suggestion
        view = ModerationRespostView(timeout=900, target_id=target_id, target_reason=moderation_reason)
```
3 minutes seems rather short for a passive mod log, 15 might be more realistic to catch?